featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -558,16 +565,18 @@ def _build_agg_features(self, all_features, parent_entity, child_entity,
             if type(input_types[0]) == list:
                 input_types = input_types[0]
 
-            features = self._features_by_type(all_features=all_features,
-                                              entity=child_entity,
-                                              max_depth=new_max_depth,
-                                              variable_type=set(input_types))
+            def feature_filter(f):
+                # Remove direct features of parent entity and features in relationship path.
+                return (not _direct_of_entity(f, parent_entity)) \
```

Comment:
wondering are both of these checks necessary? 

wouldn't `self._feature_in_relationship_path(relationship_path, f)` check if the feature is a direct of the entity?

Reply:
Maybe `_feature_in_relationship_path` should be renamed? It looks like it checks if the feature is an identity feature of a child or parent variable in the relationship path. For example, making an aggregation like `sessions: MEAN(log.session_id)` or a direct feature like `log: sessions.id` is pointless because that information is already on the target entity. That's my guess at least, although all the tests still pass if it is removed.

Reply:
ahh, you're right. let leave it as-is.